pkg/libinit/root_linux.go

```diff
@@ -284,3 +295,71 @@ func InstallModules(pattern string, exclude map[string]bool) error {
 
 	return nil
 }
+
+func readModules(f *os.File) []string {
+	r := bufio.NewReader(f)
+	modules := []string{}
+	for {
+		if i, err := r.ReadString('\n'); err == nil {
+			if strings.HasPrefix(i, "#") {
+				continue
+			} else {
+				modules = append(modules, strings.TrimSuffix(i, "\n"))
+			}
+		} else {
+			break
+		}
+	}
+	return modules
+}
+
+// InstallModules installs kernel modules from .conf files in /etc/modules-load.d/
+func InstallModulesFromModulesLoad(pattern string, exclude map[string]bool) error {
+	files, err := filepath.Glob(pattern)
+	if err != nil {
+		return err
+	}
+
+	for _, filename := range files {
+		f, err := os.Open(filename)
+		if err != nil {
+			log.Printf("installModulesFromModulesLoad: can't open %q: %v", filename, err)
+			continue
+		}
+		fModules := readModules(f)
+		for _, moduleName := range fModules {
+			if _, ok := exclude[moduleName]; ok {
+				log.Printf("Skipping module %s", moduleName)
+				continue
+			}
+			flags := cmdline.FlagsForModule(moduleName)
+			err = kmodule.Probe(moduleName, flags)
+			f.Close()
+			if err != nil {
```

Comment:
I think you can inline this:
if err := kmodule.Probe(...); err != nil {
 // error here
}

also: the function name in the log message on line 339 is missing the initial uppercase "i"